contracts/GeneratedExperience.cdc

```diff
@@ -0,0 +1,350 @@
+		) {
+			self.season = season
+			self.royalties = royalties
+			self.squareImage = squareImage
+			self.bannerImage = bannerImage
+			self.description = description
+			self.extra={}
+		}
+	}
+
+	pub struct Info {
+		pub let season: UInt64
+		pub let name: String
+		pub let description: String
+		pub let thumbnailHash: String
+		pub let fullsizeHash: String
+		pub let edition: UInt64
+		pub let maxEdition: UInt64
+		pub let artist: String
+		pub let rarity: String
+		access(self) let extra: {String: AnyStruct}
+
+		init(season: UInt64, name: String, description: String, thumbnailHash: String, edition:UInt64, maxEdition:UInt64, fullsizeHash: String, artist: String, rarity: String) {
+			self.season=season
+			self.name=name
+			self.description=description
+			self.thumbnailHash=thumbnailHash
+			self.edition=edition
+			self.maxEdition=maxEdition
+			self.fullsizeHash=fullsizeHash
+			self.artist=artist
+			self.rarity=rarity
+			self.extra={}
+		}
+	}
+
+	pub resource NFT: NonFungibleToken.INFT, MetadataViews.Resolver {
+		pub let id: UInt64
+		pub let info: Info
+
+		init(
+			info: Info
+		) {
+			self.id = self.uuid
+			self.info=info
+		}
+
+		pub fun getViews(): [Type] {
+			return [
+			Type<MetadataViews.Display>(),
+			Type<MetadataViews.Royalties>(),
+			Type<MetadataViews.Editions>(),
+			Type<MetadataViews.Traits>(),
+			Type<MetadataViews.ExternalURL>(),
+			Type<MetadataViews.NFTCollectionData>(),
+			Type<MetadataViews.NFTCollectionDisplay>(),
+			Type<MetadataViews.Medias>(),
+			Type<MetadataViews.Rarity>(),
+			Type<FindPack.PackRevealData>()
+			]
+		}
+
+		pub fun resolveView(_ view: Type): AnyStruct? {
+
+			let collection = GeneratedExperience.collectionInfo[self.info.season]!
+			let imageFile = MetadataViews.IPFSFile( cid: self.info.thumbnailHash, path: nil)
+
+			switch view {
+
+			case Type<FindPack.PackRevealData>():
+				let data : {String : String} = {
+					"nftImage" : imageFile.uri() ,
+					"nftName" : self.info.name,
+					"packType" : "GeneratedExperience"
+				}
+				return FindPack.PackRevealData(data)
+
+			case Type<MetadataViews.Display>():
+				return MetadataViews.Display(
+					name: self.info.name,
+					description: self.info.description,
+					thumbnail: MetadataViews.IPFSFile(
+						cid: self.info.thumbnailHash, path: nil
+					)
+				)
+			case Type<MetadataViews.Editions>():
+				// We do not show season here unless there are more than 1 collectionInfo (that is indexed by season)
+				var editionName = "edition"
+				if GeneratedExperience.collectionInfo.length > 1 {
+					editionName = "season ".concat(self.info.season.toString()).concat(" edition")
+				}
+				let editionInfo = MetadataViews.Edition(name: editionName, number: self.info.edition, max: self.info.maxEdition)
+				let editionList: [MetadataViews.Edition] = [editionInfo]
+				return MetadataViews.Editions(
+					editionList
+				)
+			case Type<MetadataViews.Royalties>():
+				// TODO: Do we create royalties here every time?
```

Comment:
I dont think we do, i think we send this in, and I think we should use the new fungibleTokenSwitchboard path that the flow team advocates.